admin/provisioner/clickhousestatic/provisioner.go

```diff
@@ -53,32 +52,40 @@ func New(specJSON []byte, _ database.DB, logger *zap.Logger) (provisioner.Provis
 		return nil, fmt.Errorf("failed to parse provisioner spec: %w", err)
 	}
 
-	if spec.DSNPath != "" {
-		dsn, err := os.ReadFile(spec.DSNPath)
-		if err != nil {
-			return nil, fmt.Errorf("failed to read DSN file: %w", err)
-		}
-		spec.DSN = strings.TrimSpace(string(dsn))
+	// Get primary DSN from the specified environment variable (required)
+	dsn := os.Getenv(spec.DSNEnv)
+	if dsn == "" {
+		return nil, fmt.Errorf("environment variable %s is not set or empty", spec.DSNEnv)
 	}
 
-	if spec.DSNEnv != "" && spec.DSN == "" && spec.DSNPath == "" {
-		dsn := os.Getenv(spec.DSNEnv)
-		if dsn == "" {
-			return nil, fmt.Errorf("environment variable %s is not set or empty", spec.DSNEnv)
+	// Get optional write DSN
+	var writeDSN string
+	if spec.WriteDSNEnv != "" {
+		writeDSN = os.Getenv(spec.WriteDSNEnv)
+		if writeDSN == "" {
+			return nil, fmt.Errorf("environment variable %s is not set or empty", spec.WriteDSNEnv)
 		}
-		spec.DSN = dsn
 	}
 
-	opts, err := clickhouse.ParseDSN(spec.DSN)
+	// Use writeDSN for provisioning operations if available, otherwise use the primary DSN
+	provisioningDSN := dsn
+	if writeDSN != "" {
+		provisioningDSN = writeDSN
+	}
+
+	opts, err := clickhouse.ParseDSN(provisioningDSN)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse DSN: %w", err)
 	}
 	ch := clickhouse.OpenDB(opts)
 
 	return &Provisioner{
-		spec:   spec,
-		logger: logger,
-		ch:     ch,
+		spec:     spec,
+		logger:   logger,
+		readDB:   ch,
+		writeDB:  ch,
```

Comment:
Doesn't really make sense to have two handles if they are the same? Seems like the old `p.ch` is sufficient int the provisioner (read/write separation for handles only really matter in `runtime/drivers/clickhouse`, not in the provisioner)